### PR TITLE
Add support for cohere multi-modal embeddings

### PR DESCRIPTION
**Pull request by ravi03071991**
# Description

PR to add support for cohere multi modal embeddings.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [ ] I added new unit tests to cover this change
- [ ] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py [hunk 1]
```diff
@@ -284,3 +333,24 @@ def _get_text_embeddings(self, texts: List[str]) -> List[List[float]]:
     async def _aget_text_embeddings(self, texts: List[str]) -> List[List[float]]:
         """Get text embeddings."""
         return await self._aembed(texts, input_type="search_document")
+
+    def _get_image_embedding(self, img_file_path: str) -> List[float]:
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py:337]: The input to this function is of type `ImageFileType` -- which can be string or bytesIO

File: llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py [hunk 2]
```diff
@@ -261,6 +280,36 @@ async def _aembed(self, texts: List[str], input_type: str) -> List[List[float]]:
...
+            )
+        client = self._get_client()
+        processed_image = self._image_to_base64_data_url(image_path)
+        return client.embed(
+            model=self.model_name,
+            images=[processed_image],
+            input_type=input_type,
+        ).embeddings
+
+    async def _aembed_image(self, image_path: str, input_type: str) -> List[float]:
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py:297]: This only embeds a single image at a time -- do we want to support batches? By default, `_aget_image_embeddings` and `_get_image_embeddings` will call embeddings one at a time, but we could override to do proper batch calls

File: llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py (hunk 1 above)
Comment (ravi03071991) [reply llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py:337]: Sure. I will change it accordingly.

File: llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py (hunk 2 above)
Comment (ravi03071991) [reply llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py:297]: currently does not support batches as discussed so not including it as of now.

File: llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py [hunk 3]
```diff
@@ -261,6 +281,38 @@ async def _aembed(self, texts: List[str], input_type: str) -> List[List[float]]:
...
         return getattr(result, self.embedding_type, None)
 
+    def _embed_image(self, image_path: ImageType, input_type: str) -> List[float]:
+        """Embed images using Cohere."""
+        if self.model_name not in V3_MODELS:
+            raise ValueError(
+                f"{self.model_name} is not a valid multi-modal embedding model. Supported models are {V3_MODELS}"
+            )
+        client = self._get_client()
+        processed_image = self._image_to_base64_data_url(image_path)
```
Comment (logan-markewich) [inline llama-index-integrations/embeddings/llama-index-embeddings-cohere/llama_index/embeddings/cohere/base.py:291]: If the image path is already a bytes object of the image, we might to to change how we convert that?

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**